webserver/views/api_compat.py

```diff
@@ -0,0 +1,352 @@
+from webserver.rate_limiter import ratelimit
+from webserver.errors import InvalidAPIUsage
+import xmltodict
+from api_tools import insert_payload
+from db.lastfm_user import User
+from db.lastfm_session import Session
+from db.lastfm_token import Token
+import calendar
+from datetime import datetime
+
+api_bp = Blueprint('api_compat', __name__)
+
+
+@api_bp.route('/api/auth/', methods=['GET'])
+@ratelimit()
+@login_required
+def api_auth():
+    """ Renders the token activation page.
+    """
+    token = request.args['token']
+    return render_template(
+        "user/auth.html",
+        user_id=current_user.musicbrainz_id,
+        token=token
+    )
+
+
+@api_bp.route('/api/auth/', methods=['POST'])
+@ratelimit()
+@login_required
+def api_auth_approve():
+    """ Authenticate the user token provided.
+    """
+    user = User.load_by_name(current_user.musicbrainz_id)
+    token = Token.load(request.form['token'])
+    if not token:
+        return render_template(
+            "user/auth.html",
+            user_id=current_user.musicbrainz_id,
+            msg="Either this token is already used or invalid. Please try again."
+        )
+
+    if token.api_key != user.api_key:
+        return render_template(
+            "user/auth.html",
+            user_id=current_user.musicbrainz_id,
+            msg="This token does not belong to this account. Please login in the correct account."
+        )
+    if token.user:
+        return render_template(
+            "user/auth.html",
+            user_id=current_user.musicbrainz_id,
+            msg="This token is already approved. Please check the token and try again."
+        )
+    if token.has_expired():
+        return render_template(
+            "user/auth.html",
+            user_id=current_user.musicbrainz_id,
+            msg="This token has expired. Please create a new token and try again."
+        )
+    token.approve(user.name)
+    return render_template(
+        "user/auth.html",
+        user_id=current_user.musicbrainz_id,
+        msg="Token %s approved for user %s, press continue in client." % (token.token, current_user.musicbrainz_id)
+    )
+
+
+@api_bp.route('/2.0/', methods=['POST'])
+@ratelimit()
+def api_post():
+    """ Receives the POST-API calls and redirects them to appropriate methods.
+    """
+    method = request.form['method'].lower()
+    def invalid_method_error(*args):
+        raise InvalidAPIUsage(3, output_format=request.args.get('format', "xml"))   # Invalid Method
+
+    return {
+        'track.updatenowplaying': record_listens,
+        'track.scrobble': record_listens,
+        'auth.getsession': get_session,
+        'auth.gettoken': get_token,
+        'user.getinfo': user_info
+    }.get(method, invalid_method_error)(request, request.form)
+
+
+
+def get_token(request, data):
+    """ Issue a token to user after verying his API_KEY
+    """
+    output_format = data.get("format", "xml")
+    api_key = data.get('api_key', None)
```

Comment:
`Dict.get()` returns None by default if the key doesn't exist, you dont' need to add it explicitly.

Be careful about using different quotes in the same file. Pick one and stick with it (we tend to use double-quotes)